patches/minecraft/net/minecraft/client/gui/screens/Screen.java.patch

```diff
@@ -165,14 +168,27 @@
     }
  
     public boolean m_7043_() {
-@@ -570,6 +_,10 @@
+@@ -572,6 +_,10 @@
+    public void m_7400_(List<Path> p_96591_) {
     }
  
-    public void m_7400_(List<Path> p_96591_) {
-+   }
-+
 +   public Minecraft getMinecraft() {
 +      return this.f_96541_;
-    }
- 
++   }
++
     private void m_169380_(long p_169381_, boolean p_169382_) {
+       this.f_169377_ = Util.m_137550_() + p_169381_;
+       if (p_169382_) {
+@@ -734,4 +_,12 @@
+          this.f_169422_ = p_169426_;
+       }
+    }
++
++   private void addEventWidget(GuiEventListener b) {
```

Comment:
Shouldn't this method handle the three allowed cases by vanilla?

Method | Renderable | Narratable | Children
:---: | :---: | :---: | :---:
addRenderableWidget | 1 | 1 | 1
addRenderableOnly | 1 | 0 | 0
addWidget | 0 | 1 | 1

Reply:
What would the usecases of this be?
And if we did this we would have to add these mirror methods to the event class itself.
This function is there because we don't want to expose access to inner fields.

Reply:
The main issue is that pretty much everything extends `Renderable` and `NarratableEntry` such that everything would be essentially be delegated to `#addRenderableWidget`. So, it's difficult to perform any kind of granularity of if you want to use a preexisting widget but want it only to display.

However, you could also argue vanilla is bad at this since their logic could be front loaded into the widget itself. So, honestly, you can probably just ignore this comment.

Reply:
Given how old this event is and that we haven't had any requests for a different method of widget addition, I think we can call it a non-issue.  If this was desired functionality surely _someone_ would have mentioned something.

Reply:
I'm of the same opinion, if someone wants it, they can ask for it and pull it in later.
This is no reason to delay what is essentially a bug fix.